Makefile.helper.mk

```diff
@@ -46,16 +46,6 @@ e2e-test:
           KUBERNETES_CONFIG="$(KUBECONFIG)" go test -v -timeout 40m ./test/e2e/$$d -ginkgo.v -ginkgo.noColor -ginkgo.failFast || exit; \
         done
 
-# Download kube-linter locally if necessary
-KUBELINTER = $(shell pwd)/bin/kube-linter
```

Comment:
I think this comes straight from the `Makefile` that operator-sdk generates. Do we want to diverge from it again?

Reply:
Can you elaborate on how it was generated and why we needed that generation?

Reply:
Sure!
I understand that the operator was created using the [Operator Framework](https://sdk.operatorframework.io/).
The framework includes `operator-sdk`, which [generates the initial Makefile](https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/#steps).
When upgrading to a newer version of the framework, it is nice to have the Makefile as close to the original one as possible, as the [upgrade guide](https://sdk.operatorframework.io/docs/upgrading-sdk-version/) explains what changes should be made to it. Many changes to the Makefile are what made #116 quite painful.

Reply:
Ummm last time I checked (long time ago) the operator-sdk wasn't the recommended tool for operators (I may be wrong and it may have changed).

1. How do we expect to use operator-sdk in the future?
2. I don't see any change log for using `go install` instead of the deprecated `go get` for binaries - can you point me to that direction? @qbarrand 

@mfilanov do we use `operator-sdk` in assisted?

Reply:
Is there another framework that we should use?

1. I can imagine that we will use some features at some point, and by using it we remain aligned with other operators that also use the framework.
2. Indeed, I ran `operator-sdk init` with the latest version and it still calls `go get`. But as long as it works, I would try not to touch the Makefile too much.

Reply:
This is what we install 
1. https://github.com/openshift/assisted-service/blob/fb2c51618d44bf6ed3855b1ad53c1ddbbcf1dc7c/hack/setup_env.sh#L57-L62
2. https://github.com/openshift/assisted-service/blob/fb2c51618d44bf6ed3855b1ad53c1ddbbcf1dc7c/hack/setup_env.sh#L70
not sure what is the operator used for, but to generate the CRDs we are using controller-gen https://github.com/openshift/assisted-service/blob/fb2c51618d44bf6ed3855b1ad53c1ddbbcf1dc7c/hack/generate.sh#L150

Reply:
This is the operator sdk - https://github.com/openshift/assisted-service/blob/fb2c51618d44bf6ed3855b1ad53c1ddbbcf1dc7c/hack/setup_env.sh#L59 @filanov

Reply:
@mhrivnak maybe you have some insights about it?

tl;dr:
1. Isn't it "unexpected" from the user side that using random makefile target, e.g. `make lint`, is installing binaries on the machine?
2. using `go get` for binaries is deprecated, does operator-sdk have a plane to replace those with `go install`?

Reply:
operator-sdk is the recommended way to create operators. It extends [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/), which is responsible for scaffolding things like the [Makefile](https://github.com/kubernetes-sigs/kubebuilder/blob/57688c578b5d1189a7a1f28ebd1870ac9b534cd9/pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go).

One of the operator-sdk maintainers [made a PR to kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/pull/2383) converting to `go install`.

Reply:
Ack. Thanks!